### ISSUE TITLE
DPI Assert Failure on connecting new monitor

### ISSUE THREAD

**Issue by Jon-Admiral-Instruments** [ocornut/imgui#6594]
```
Dear ImGui 1.89.7 (18971)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=199711
define: IMGUI_DISABLE_OBSOLETE_FUNCTIONS
define: IMGUI_DISABLE_OBSOLETE_KEYIO
define: _WIN32
define: _WIN64
define: _MSC_VER=1935
define: _MSVC_LANG=201703
define: IMGUI_HAS_VIEWPORT
define: IMGUI_HAS_DOCK
--------------------------------
io.BackendPlatformName: imgui_impl_glfw
io.BackendRendererName: imgui_impl_vulkan
io.ConfigFlags: 0x00000441
 NavEnableKeyboard
 DockingEnable
 ViewportsEnable
io.ConfigViewportsNoAutoMerge
io.ConfigViewportsNoDecoration
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x00001C0E
 HasMouseCursors
 HasSetMousePos
 PlatformHasViewports
 HasMouseHoveredViewport
 RendererHasVtxOffset
 RendererHasViewports
--------------------------------
io.Fonts: 10 fonts, Flags: 0x00000000, TexSize: 2048,4096
io.DisplaySize: 1280.00,720.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

The following assert triggers when connecting a specific monitor (opening laptop screen). The same assert does not apply (and the DpiScale is 1.5x) if the monitor is already on at program start. Without the assert, no program issues are observed, but DpiScale does remain 0.

```            IM_ASSERT(mon.DpiScale != 0.0f);```
In function "static void ImGui::ErrorCheckNewFrameSanityChecks()"

Reproducibility may be an issue, as it only happens with one of the laptops I have tested.

Debugging reveals that this error is triggered

```
if (IsWindows8Point1OrGreater())
    {
        if (GetDpiForMonitor(handle, MDT_EFFECTIVE_DPI, &xdpi, &ydpi) != S_OK)
        {
            _glfwInputError(GLFW_PLATFORM_ERROR, "Win32: Failed to query monitor DPI");
            return;
        }
    }
```

Root cause would seem to be GLFW or windows and a not fully initialized monitor, but would it be more appropriate to assume a DpiScale of 1 rather than assert? Is anyone else having this issue?

**Comment by ocornut** [comment]
I would suggest you start by debugging it by modifying the code in `ImGui_ImplGlfw_UpdateMonitors()`:

The code is:
```cpp
#if GLFW_HAS_PER_MONITOR_DPI
        // Warning: the validity of monitor DPI information on Windows depends on the application DPI awareness settings, which generally needs to be set in the manifest or at runtime.
        float x_scale, y_scale;
        glfwGetMonitorContentScale(glfw_monitors[n], &x_scale, &y_scale);
        monitor.DpiScale = x_scale;
#endif
```

(1) Add
```cpp
#include "imgui_internal.h"
...
IMGUI_DEBUG_LOG("Monitor %d dpi_scale %f\n", n, monitor.DpiScale);
```
This will record the frame count as well.

(2)
Try 
```cpp
if (monitor.DpiScale == 0.0f)
{
   monitor.DpiScale = 1.0f;
   bd->WantUpdateMonitors = true; // try again next frame
}
```
And see first if the value gets better on subsequent frames (or not) ? If it does we can decide on better code to do that.

**Comment by Jon-Admiral-Instruments** [comment]
I tried the above.
The monitor showed up as 0 DpiScale for every subsequent frame. After a few thousand log messages windows bluescreened and restarted.

I ran a windows disk scan and repair, reinstalled and updated all my graphics drivers, and am no longer able to observe the issue.

Thank you for your time, it was likely a windows or graphics driver error. I am marking the issue closed.
